client/src/modules/members/model/actions.js

```diff
@@ -0,0 +1,74 @@
+import _keyBy from 'lodash/keyBy';
+
+import { MembersActionTypes } from './actionTypes';
+import { getData, patchData } from 'common/utils/fetchMethods';
+import { createNotificationWithTimeout } from 'modules/notification/model/actions';
+import { MessageType as NotificationType } from 'common/constants/enums';
+import { ENDPOINT_URL } from 'common/constants/variables';
+
+const fetchMembersRequest = () => ({
+  type: MembersActionTypes.FETCH_REQUEST
+});
+
+const fetchMembersFailure = () => ({
+  type: MembersActionTypes.FETCH_FAILURE
+});
+
+const fetchMembersSuccess = data => ({
+  type: MembersActionTypes.FETCH_SUCCESS,
+  payload: data
+});
+
+const clearMembersRequest = () => ({
+  type: MembersActionTypes.CLEAR_DATA
+});
+
+const addMemberRequest = () => ({
+  type: MembersActionTypes.ADD_REQUEST
+});
+
+const addMemberSuccess = data => ({
+  type: MembersActionTypes.ADD_SUCCESS,
+  payload: data
+});
+
+const addMemberFailure = () => ({
+  type: MembersActionTypes.ADD_FAILURE
+});
+
+export const fetchCohortMembers = cohortId => dispatch => {
+  dispatch(fetchMembersRequest());
+  return getData(`${ENDPOINT_URL}/cohorts/${cohortId}/get-members`)
+    .then(resp => resp.json())
+    .then(json => {
+      const data = _keyBy(json, '_id');
+      dispatch(fetchMembersSuccess(data));
+    })
+    .catch(err => {
+      dispatch(fetchMembersFailure());
+      createNotificationWithTimeout(
+        dispatch,
+        NotificationType.ERROR,
+        err.message || "Oops, we're sorry, fetching members failed..."
+      );
+    });
+};
+
+export const clearMembers = () => dispatch => dispatch(clearMembersRequest());
```

Comment:
1. I'm not sure that we have to dispatch clearMemberRequest() - we simply can invoke this action creator directly like in list's case:
```
export const removeArchivedListsMetaData = () => ({
  type: ListActionTypes.REMOVE_ARCHIVED_META_DATA
});
```
2. This action creator doesn't have to have request in it's name because there is no request here. If you want to leave this additional word change it to success, because on success we invoke actions which purpose is to modify store.